zerver/lib/user_groups.py

```diff
@@ -558,8 +559,8 @@ def get_setting_value_for_user_group_object(
 
 def get_group_setting_value_for_register_api(
     setting_group_id: int,
-    realm_setting_anonymous_group_membership: dict[int, UserGroupMembersDataDict],
-) -> int | UserGroupMembersDataDict:
+    realm_setting_anonymous_group_membership: dict[int, UserGroupMembersTypedDict],
+) -> int | UserGroupMembersTypedDict:
```

Comment:
This function was only being used at one place and it was not making use of the benefits that a dataclass provides. So replacing it with a typed dict seemed fine.